### ISSUE TITLE
H5P-activity that has multiple videos doesn't open at all, server times out.

### ISSUE THREAD

**Issue by Jarvil** [h5p/h5p-moodle-plugin#160]
Hello,

if you have course presentation that has multiple video files the h5p-activity won't open at all because Moodle server times out. 

Moodle ; 3.1.5

**Comment by Jarvil** [comment]
It takes over 240 seconds to complete.

**Comment by Jarvil** [comment]
Have anyone tested f.ex. course presentation in Moodle that has many videofiles?

**Comment by Jarvil** [comment]
It seems that H5P is trying to unpack to server TEMP-directory before publishing it in Moodle and this causes the timeout.

**Comment by Jarvil** [comment]
Anyone?

**Comment by timothyylim** [comment]
Hi @Jarvil , sorry I haven't gotten round to this. I'll reply within the next day.

**Comment by timothyylim** [comment]
Hi @Jarvil, I've made an issue for it in our issue tracker:

https://h5ptechnology.atlassian.net/browse/HFP-1372

It should be part of the next release.

**Comment by timothyylim** [comment]
Thanks for taking the time to report it

**Comment by thomasmars** [comment]
@Jarvil That will naturally take extra long time to process. Have you tried to increase the max timeout settings of your server ?
What happens if you try to visit the activity after it timed out ? Did the activity get created ? What happens ?

**Comment by Jarvil** [comment]
@thomasmars Increasing max timeout on the server will eventually let me open the H5P-presentation but this cannot be an permanent solution since it isn't wise to keep too long max timeout values on the server.

The problem happens when you edit existing H5P-activity that has multiple videos, it seems that Chrome is more likely to display the error or crash completely than f.ex. Internet Explorer 11 or Firefox (Maybe Chrome can maintain lesser amount of concurrent connections at the same time.)

Problem occurs when you try to edit H5P-activity with multiple video files on Moodle course, it will just load and load and after loading for minutes it will display an error;

"Timed out while waiting for session lock. Wait for your current requests to finish and try again later."

**Comment by falcon-git** [comment]
If you disable export for these nodes it will probably work without a problem. I believe it is creating the H5P file - moving and zipping all that data - that takes time.

Course Presentation loads the first slide and the next slide, and not the rest until you navigate IIRC, so it shouldn't have a problem with displaying the content.

We probably should check file sizes and skip the export, do the export on cron, include links to the videos instead of packing them or some other solution if it is getting huge. What do you think?

**Comment by Jarvil** [comment]
@falcon-git  Hello, that zipping and packing might be the key issue. Coudn't you just skip the zipping/packaging alltogether and do it only if someone presses "download" button on the H5P? That is what zipping/packing in the background is created for right?

**Comment by falcon-git** [comment]
Good point. We do the zipping once and store it for performance reasons, but it might be a miscalculation there since people probably don't download that often.

When the Content Hub is up and running this year we will probably stop creating those in advance. Maybe we'll do it sooner. I'm thinking about it. Good feedback

**Comment by Jarvil** [comment]
Hello,

Any progress on this @falcon-git It is still easy to "brick"/make it so you can't open HP5-activity at all if you use H5P course-presentation library with big videos in it. We are talking about mod_hvp 1.5

**Comment by falcon-git** [comment]
Glad you bump this one. The Hub is significantly delayed so we should fix this sooner.
@Jarvil I've created this issue: https://h5ptechnology.atlassian.net/browse/HFP-1731

Did you try to disable download on your site?

**Comment by Jarvil** [comment]
Hello,

I haven't been able to replicate the problem myself yet but soon as I'm able I will try to disabling the download-option.

**Comment by falcon-git** [comment]
Update: we've fixed this in core. Will be added to Moodle and the other plugins soon.

**Comment by Jarvil** [comment]
Hello,

Can you specify in which mod_hvp version was this problem fixed? And why is this still open:

https://h5ptechnology.atlassian.net/browse/HFP-1731

@falcon-git

**Comment by falcon-git** [comment]
@Jarvil It is still open since it hasn't been ported to all the publishing systems yet. It might unfortunately take quite some time before we're able to do that. It is on the export on demand branch of the h5p-php-library

**Comment by haietza** [comment]
We experienced the issue of a possibly large video file being uploaded into the Interactive Video content type causing the PHP Fatal error:  Maximum execution time of 30 seconds exceeded with the Moodle plugin. It caused several large hvp-[uniqueid] files to be left in the moodledata temp directory that then caused the disk to run out of space and take the site down. What is the status of this fix being pushed to the plugin please? Does disabling export in the plugin settings work as a temporary workaround for this issue?
